src/components/HorizontalCard.jsx

```diff
@@ -35,28 +31,28 @@ export function HorizontalCard(props) {
                         //if the quantity is 1 then removes the product
                         //instead of showing quantity in 0 or negative number
                         if(props.quantity === 1) {
-                            deleteFromCart(token, props.productId)
+                            cartDispatch({type: "DELETE-FROM-CART", payload: props.productId})
                         }
                         else {
-                            updateCartQuantity(token, props.productId, "decrement")}
+                            cartDispatch({type: "UPDATE-CART-QUANTITY", payload: props.productId, updateType: "decrement"})
                         }
-                    }>-</button> 
+                    }}>-</button> 
 
                     <span>{props.quantity}</span> 
 
-                    <button onClick={() => updateCartQuantity(token, props.productId, "increment")}>+</button>
+                    <button onClick={() => cartDispatch({type: "UPDATE-CART-QUANTITY", payload: props.productId, updateType: "increment"})}>+</button>
                 </span></div>
             
                 <div className="product-action">
-                    <button className="btn btn-yellow" onClick={() => deleteFromCart(token, props.productId)}>Remove from Cart</button>
+                    <button className="btn btn-yellow" onClick={() => cartDispatch({type: "DELETE-FROM-CART", payload: props.productId})}>Remove from Cart</button>
```

Comment:
in your dispatch type has a value which is a constant it should follow constant naming convention.
DELETE-FROM-CART => DELETE_FROM_CART